### ISSUE TITLE
Last build step fails to install shiboken2.so on OS X

### ISSUE THREAD

**Issue by ctismer** [PySide/pyside2#58]
When building PySide2 with these settings:
```
python3 setup.py build --debug --no-examples --ignore-git --qmake=/usr/local/Cellar/qt5/5.5.1/bin/qmake --build-tests --jobs=9
```
everything builds, but then the install fails with
```
Copying file /Users/tismer/src/pyside2-setup/pyside_install/py3.4-qt5.5.1-64bit-debug/lib/python3.4/site-packages/shiboken2.so to /Users/tismer/src/pyside2-setup/pyside_package/PySide2/shiboken2.so.
error: [Errno 2] No such file or directory: '/Users/tismer/src/pyside2-setup/pyside_install/py3.4-qt5.5.1-64bit-debug/lib/python3.4/site-packages/shiboken2.so'
```
Checking if the files exist:
```
ls -la /Users/tismer/src/pyside2-setup/pyside_install/py3.4-qt5.5.1-64bit-debug/lib/python3.4/site-packages/shiboken2*
ls: /Users/tismer/src/pyside2-setup/pyside_install/py3.4-qt5.5.1-64bit-debug/lib/python3.4/site-packages/shiboken2*: No such file or directory
```

**Comment by thopiekar** [comment]
Looks like Shiboken2 does not install the library.

What does ```ls -la /Users/tismer/src/pyside2-setup/pyside_install/py3.4-qt5.5.1-64bit-debug/lib/python3.4/site-packages/``` say?
And ```find /Users/tismer/src/pyside2-setup | grep shiboken2.so```?

**Comment by ctismer** [comment]
```
ls -la /Users/tismer/src/pyside2-setup/pyside_install/py3.4-qt5.5.1-64bit-debug/lib/python3.4/site-packages/
total 0
drwxr-xr-x   4 tismer  501  136 Jan 21 00:19 .
drwxr-xr-x   3 tismer  501  102 Jan 21 00:18 ..
drwxr-xr-x  27 tismer  501  918 Jan 31 16:56 PySide2
drwxr-xr-x  14 tismer  501  476 Jan 31 16:56 pysideuic
```

```
find /Users/tismer/src/pyside2-setup | grep shiboken2.so
/Users/tismer/src/pyside2-setup/.git/modules/sources/shiboken2/sourcetreeconfig
/Users/tismer/src/pyside2-setup/pyside_build/py3.4-qt5.5.1-64bit-debug/shiboken2/shibokenmodule/shiboken2.so
```

**Comment by ctismer** [comment]
arrgh - do you think it tries to install shiboken2 in the real install dir, and then has no permission?

Oh well, that would explain it. I'm always building with no virtualenv or sudo.

**Comment by thopiekar** [comment]
Well, I don't know whether your problem is related to my recent changes. I hoped the library has only a wrong suffix. However, I wonder why libshiboken2.so should be in site-packages.

Here on Ubuntu libshiboken is at:
```
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libshiboken-python2.7.so.1.2
/usr/lib/x86_64-linux-gnu/libshiboken-python2.7.so.1.2.2
```

and in my shiboken2 packages at:
```
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libshiboken2-python2.7.so.2.0
/usr/lib/x86_64-linux-gnu/libshiboken2-python2.7.so.2.0.0
```

Which is correct here. Is libshiboken in any way a python module so it should be at site-packages?

**Comment by thopiekar** [comment]
When I clone pyside2-setup and run ```python setup.py install```, where will the installation go? Is there a way to set DESTDIR=./tmp? Just don't want to have different files lying around.

**Comment by ctismer** [comment]
It will install into the site-packages of that Python.

**Comment by thopiekar** [comment]
Is there a way to set an alternative path?

**Comment by IronManMark20** [comment]
So I think @ctismer is right, it isn't being copied. My Install just failed because it couldn't copy.

    -- Installing: C:/Users/--username--/Documents/pyside2-setup/pyside_install/py3.5-qt5.5.1-32bit-release/bin/shiboken2.exe
    -- Installing: C:/Program Files (x86)/Python 3.5/Lib/site-packages/shiboken2.pyd
    CMake Error at shibokenmodule/cmake_install.cmake:39 (file):
      file INSTALL cannot copy file "C:/Users/--user
      name--/Documents/pyside2-setup/pyside_build/py3.5-qt5.5.1-32bit-release/shiboken2/shibokenmodule/shiboken2.pyd"
      to "C:/Program Files (x86)/Python 3.5/Lib/site-packages/shiboken2.pyd".
    Call Stack (most recent call first):
      cmake_install.cmake:36 (include)
    
    
    NMAKE : fatal error U1077: 'C:\cmake\bin\cmake.exe' : return code '0x1'
    Stop.
    error: Error pseudo installing shiboken2

**Comment by thopiekar** [comment]
That's sad :disappointed:
Are there any shiboken files at C:/Program Files (x86)/Python 3.5/Lib/site-packages/ ?

**Comment by IronManMark20** [comment]
Hm. There is an 11kb shiboken2.pyd. So, maybe it fails part way through? Or it fails due to an existing file? I will try removing it and starting over

**Comment by thopiekar** [comment]
That's strange. Isn't the default behavior of CMake to "update" a file, if it exists?

**Comment by IronManMark20** [comment]
Yeah, very strange. I am guessing that since in Py3.5, the files are in Program Files (x86), and since there are admin restrictions, you need admin to do anything. It seems if I remove the shiboken2.pyd, and run from admin command prompt, things are fine.

**Comment by ctismer** [comment]
In any case, I'm using python 3.4.3.
I just evicted any traces of older shiboken installs and will try, again.

**Comment by thopiekar** [comment]
@IronManMark20 I always worry when using such scripts gaining root-rights :smile:

**Comment by ctismer** [comment]
bad bad bad :-D

**Comment by ctismer** [comment]
Crazy!!
It worked, after I removed an old PySide installation!
So my theory is that some things are guessed and influenced wrongly by
an old PySide install ?!

**Comment by ctismer** [comment]
```
find . /usr/local/lib/python3.4/site-packages/ -name 'shiboken*.so'
./build/lib.macosx-10.6-intel-3.4/PySide2/shiboken2.so
./pyside_build/py3.4-qt5.5.1-64bit-debug/shiboken2/shibokenmodule/shiboken2.so
./pyside_install/py3.4-qt5.5.1-64bit-debug/lib/python3.4/site-packages/shiboken2.so
./pyside_package/PySide2/shiboken2.so
```
This works, after PySide is evicted from the machine. I think the confusion comes when some shiboken is already existing.

**Comment by thopiekar** [comment]
Sounds plausible. But a shame we didn't catch the reason for it. However, good to know it is "fixed".

While reading the setup.py I found a interesting section of the code:
https://github.com/PySide/pyside2-setup/blob/master/setup.py#L164

Wouldn't it make sense to add a last "if"-clause, which checks for the existence of the executable?
Eg, 
```
if not os.path.isfile(OPTION_CMAKE):
    if OPTION_CMAKE:
        print("Warning: Executable %s given by OPTION_CMAKE does not exist!" %(OPTION_CMAKE))
    else:
        print("Warning: OPTION_CMAKE not set!")
    sys.exit(1)
```

In case qmake is not found the script would still run with ```OPTION_CMAKE = None```.

**Comment by IronManMark20** [comment]
I can confirm that it is the shiboken2.pyd pre-existing btw, I'll create an issue for it.

**Comment by ctismer** [comment]
and I can't find a reason why shiboken should influence shiboken2 ???

thosetup.py: yes, please! I always found this file huge and a bit wrong.
It does too much in too many sections.

**Comment by thopiekar** [comment]
:+1:

**Comment by thopiekar** [comment]
@IronManMark20 So it needs a if os.path.isfile(pyd): os.remove(pyd)
Or the tool being used here has just a "update files" function disabled.

**Comment by IronManMark20** [comment]
See #59
